### PR TITLE
MAILBOX-342 Improve CassandraModule declarations

### PR DESCRIPTION
**Pull request by chibenwa**
Use the stage builder introduced in https://github.com/linagora/james-project/pull/1557

### CODE REVIEW

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionModule.java [hunk 1]
```diff
@@ -22,39 +22,16 @@
...
-    }
-
-    @Override
-    public List<CassandraType> moduleTypes() {
-        return ImmutableList.of();
-    }
+public interface CassandraSchemaVersionModule {
+    CassandraModule MODULE = CassandraModule.table(CassandraSchemaVersionTable.TABLE_NAME)
+        .statement(statement -> statement
+            .ifNotExists()
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionModule.java]: what about putting this `ifNotExists` into the builder ? It looks more like a policy than a definition, don't you think ?

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionModule.java [hunk 2]
```diff
@@ -22,39 +22,16 @@
...
+            .addPartitionKey(CassandraSchemaVersionTable.KEY, timeuuid())
+            .addClusteringColumn(CassandraSchemaVersionTable.VALUE, cint())
+            .withOptions()
+            .comment("Holds the history of the versions of the schema used."))
```
Comment (mbaechler) [inline backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionModule.java]: what about enforcing the comment using the CassandraModule builder ?

File: server/data/data-cassandra/src/main/java/org/apache/james/domainlist/cassandra/CassandraDomainListModule.java [hunk 3]
```diff
@@ -21,41 +21,20 @@
...
-    @Override
-    public List<CassandraTable> moduleTables() {
-        return tables;
-    }
 
-    @Override
-    public List<CassandraType> moduleTypes() {
-        return types;
-    }
+public class CassandraDomainListModule {
```
Comment (mbaechler) [inline server/data/data-cassandra/src/main/java/org/apache/james/domainlist/cassandra/CassandraDomainListModule.java]: interface ?

File: server/data/data-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepositoryUrlModule.java [hunk 4]
```diff
@@ -21,41 +21,19 @@
...
-    @Override
-    public List<CassandraTable> moduleTables() {
-        return tables;
-    }
 
-    @Override
-    public List<CassandraType> moduleTypes() {
-        return types;
-    }
+public class CassandraMailRepositoryUrlModule {
```
Comment (mbaechler) [inline server/data/data-cassandra/src/main/java/org/apache/james/mailrepository/cassandra/CassandraMailRepositoryUrlModule.java]: interface ?

File: server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRRTModule.java [hunk 5]
```diff
@@ -21,44 +21,22 @@
...
-    @Override
-    public List<CassandraTable> moduleTables() {
-        return tables;
-    }
 
-    @Override
-    public List<CassandraType> moduleTypes() {
-        return types;
-    }
+public class CassandraRRTModule {
```
Comment (mbaechler) [inline server/data/data-cassandra/src/main/java/org/apache/james/rrt/cassandra/CassandraRRTModule.java]: interface ?

**Comment by chibenwa** [comment]
We can also add `modules(...)` support to the builder and drop `CassandraCompositeModule`

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionModule.java (hunk 1 above)
Comment (chibenwa) [reply backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionModule.java]: I hesitated about this, but I clearly do agree with you.

File: backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionModule.java (hunk 2 above)
Comment (chibenwa) [reply backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/versions/CassandraSchemaVersionModule.java]: Nice idea!

**Comment by chibenwa** [comment]
Failed while dockering JPA Guice. Due to weak dockerfile isolation (this PR isNOT rebased on master hence incorrect JPA java-agant version is used...) Can be considered :green_apple:

**Pull request closed without merging**